From da4554af04a79d2be1ede0d30fb70b3e3a2881cd Mon Sep 17 00:00:00 2001
From: Thomas Graf <tgraf@suug.ch>
Date: Fri, 1 Feb 2013 10:21:58 +0100
Subject: [PATCH] cache: Make NL_ACT_* and nl_cache_ops_(get|put)() available

Move NL_ACT_* definition and the functions nl_cache_ops_get()
and nl_cache_ops_put() into the public facing API. They can
be considered stable.

Signed-off-by: Thomas Graf <tgraf@suug.ch>
---
 include/netlink-private/cache-api.h | 15 ---------------
 include/netlink/cache.h             | 15 +++++++++++++++
 2 files changed, 15 insertions(+), 15 deletions(-)

diff --git a/include/netlink-private/cache-api.h b/include/netlink-private/cache-api.h
index 5d9b0ad..f3d9f01 100644
--- a/include/netlink-private/cache-api.h
+++ b/include/netlink-private/cache-api.h
@@ -109,18 +109,6 @@ extern "C" {
  * @{
  */
 
-enum {
-	NL_ACT_UNSPEC,
-	NL_ACT_NEW,
-	NL_ACT_DEL,
-	NL_ACT_GET,
-	NL_ACT_SET,
-	NL_ACT_CHANGE,
-	__NL_ACT_MAX,
-};
-
-#define NL_ACT_MAX (__NL_ACT_MAX - 1)
-
 #define END_OF_MSGTYPES_LIST	{ -1, -1, NULL }
 
 /**
@@ -273,9 +261,6 @@ struct nl_cache_ops
 	struct nl_msgtype	co_msgtypes[];
 };
 
-extern void	nl_cache_ops_get(struct nl_cache_ops *);
-extern void	nl_cache_ops_put(struct nl_cache_ops *);
-
 /** @} */
 
 #ifdef __cplusplus
diff --git a/include/netlink/cache.h b/include/netlink/cache.h
index 0da1377..415bb00 100644
--- a/include/netlink/cache.h
+++ b/include/netlink/cache.h
@@ -21,6 +21,18 @@
 extern "C" {
 #endif
 
+enum {
+	NL_ACT_UNSPEC,
+	NL_ACT_NEW,
+	NL_ACT_DEL,
+	NL_ACT_GET,
+	NL_ACT_SET,
+	NL_ACT_CHANGE,
+	__NL_ACT_MAX,
+};
+
+#define NL_ACT_MAX (__NL_ACT_MAX - 1)
+
 struct nl_cache;
 typedef void (*change_func_t)(struct nl_cache *, struct nl_object *, int, void *);
 
@@ -148,6 +160,9 @@ extern void			nl_cache_mngr_info(struct nl_cache_mngr *,
 						   struct nl_dump_params *);
 extern void			nl_cache_mngr_free(struct nl_cache_mngr *);
 
+extern void			nl_cache_ops_get(struct nl_cache_ops *);
+extern void			nl_cache_ops_put(struct nl_cache_ops *);
+
 #ifdef __cplusplus
 }
 #endif
-- 
1.8.1.2

